src/server/extensions/compatibility-layer/searx/bridge/engine.py

```
@@ -0,0 +1,128 @@
+    spec = importlib.util.spec_from_file_location(name, path)
+    if not spec or not spec.loader:
+        raise RuntimeError("could not load engine")
+    mod = importlib.util.module_from_spec(spec)
+    sys.modules[name] = mod
+    spec.loader.exec_module(mod)
+    return mod
+
+
+def _adopt(mod):
+    if not hasattr(mod, "logger"):
+        setattr(mod, "logger", Logger())
```

Ruff (0.16.0)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)